conans/cli/api/subapi/graph.py

```diff
@@ -9,59 +10,63 @@ class GraphAPI:
             raise ConanException("Both path and reference arguments were provided. Please provide "
                                  "only one of them")
 
         app = ConanApp(self.conan_api.cache_folder)
+        # necessary for correct resolution and update of remote python_requires
+        remote = [remote] if remote is not None else None
+        app.load_remotes(remote, update=update)
+
+        graph_manager, cache = app.graph_manager, app.cache
+        reference = reference or path
+        root_node = graph_manager._load_root_node(reference, create_reference, profile_host,
+                                                  lockfile, root_ref, is_build_require,
+                                                  require_overrides)
+        return root_node
+
+    # Maybe we don't want to pass Profile objects to the API?
+    @api_method
+    def load_graph(self, root_node, profile_host, profile_build, lockfile,
+                   remote=None, update=False):
+        """ Calculate graph and fetch needed recipes
+        """
+        app = ConanApp(self.conan_api.cache_folder)
 
         remote = [remote] if remote is not None else None
         app.load_remotes(remote, update=update)
 
         assert profile_host is not None
         assert profile_build is not None
 
-        if path and reference:
-            raise ConanException("Both path and reference arguments were provided. Please provide "
-                                 "only one of them")
-
-        graph_manager, cache = app.graph_manager, app.cache
-
         out = ConanOutput()
-        out.info("Configuration (profile_host):")
-        out.info(profile_host.dumps())
-        out.info("Configuration (profile_build):")
-        out.info(profile_build.dumps())
 
-        ref_or_path = reference or path
-        deps_graph = graph_manager.load_graph(ref_or_path, create_reference, profile_host,
-                                              profile_build,
-                                              lockfile, root_ref, build_modes,
-                                              is_build_require=is_build_require,
-                                              require_overrides=require_overrides)
-
-        deps_graph.report_graph_error()
+        builder = DepsGraphBuilder(app.proxy, app.loader, app.range_resolver)
+        deps_graph = builder.load_graph(root_node, profile_host, profile_build, lockfile)
+        # FIXME: this ugly ouput doesn't belong here
+        version_ranges_output = app.range_resolver.output
+        if version_ranges_output:
+            out.success("Version ranges solved")
+            for msg in version_ranges_output:
+                out.info("    %s" % msg)
+            out.writeln("")
+            app.range_resolver.clear_output()
 
         if lockfile:
             lockfile.update_lock(deps_graph)
 
         return deps_graph
+
+    @api_method
+    def analyze_binaries(self, deps_graph, build_mode=None, remote=None, update=None):
```

Comment:
Docstring

Reply:
Done, in this and other methods too, completing params.
Some that are still too unstable, and very unlikely to remain, I have put TODOs.